### PR TITLE
Expense App Final

### PR DESCRIPTION
**Pull request by ginabeki**
In this project, I did the following:-

**Splash screen**

- A simple page with the name of my app, and links to the signup and log-in pages.
- Created signup and login pages
- The user should be able to register in the app with full name, email, and password (all mandatory).
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

**Home page (categories page)**

- When the user logs in, they are presented with the categories page.
- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

**Transactions page**

- For a given category, the list of transactions is presented, ordered by the most recent.
- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

 **"Add a new category" page**
- 
- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

 **"Add a new transaction" page**

- The user fills out a form to create a new transaction with:
- name (mandatory)
- amount (mandatory)
- categories (mandatory at least one)
- The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,25 @@
 class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+  # before_action :authenticate_user!, except: [:splash], only: [:index]
```
Comment (Rustamxon7) [inline app/controllers/application_controller.rb:3]: **You did great job!**

- [ ] To make your code clean, please remove unused commented codes from it.

File: README.md [hunk 2]
```diff
@@ -1,24 +1,156 @@
...
 
-* Services (job queues, cache servers, search engines, etc.)
+- User Registration.
+- User login.
+- Create Category: a user can create a category for his expenses.
+- Create transaction(expense): a user can create a transaction that belongs to a specific category.
 
-* Deployment instructions
+## 🚀 Live Demo <a name="live-demo"></a>
+- [Live Demo](https://watch.screencastify.com/v/MI0yFFvEqTKWdHVUP4X9)
```
Comment (Rustamxon7) [inline README.md:53]: - [ ] Perfect work on your project. However, I can see that you have issue in your deployment. Please fix it up, and resubmit it again.

**Review by Rustamxon7 — CHANGES_REQUESTED** [review]
Hello @ginabeki,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_
_Best implemented tests_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rustam-tolipov** [comment]
Hi @ginabeki, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that needed to include a future features section in the readme.

Please note that all projects must include a Future Features section outlining 1-3 potential features to be added. You have submitted your project without this section, so please add it before submitting again. Thank you for your understanding.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/views/expenses/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<h1 class="uppercase text-center text-3xl">Add Transaction</h1>
+<%= form_with model: @expense, html: { class: "flex flex-col gap-4 m-8 items-center" }, url: expenses_path, scope: :expense do |f|%>
+
```
Comment (OLIPLICHE) [inline app/views/expenses/new.html.erb:3]: - Well done 👍 
Capture
-

Capture1

**Review by OLIPLICHE — APPROVED** [review]
Hi @ginabeki, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- The App has been deployed and working well 👍 
- Used Potgress as a database 👍 
- Has Unit and Integration test 👍 
- Test has passed 👍 
- Linter is green (the last commit has a green check mark (✅)) 👏
- README file is descriptive 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**